cpp/ql/test/README.md

```diff
@@ -0,0 +1,33 @@
+# C/C++ CodeQL tests
+
+This document provides additional information about the C/C++ CodeQL Tests located in `cpp/ql/test`.  See [Contributing to CodeQL](CONTRIBUTING.md) for general information about contributing to this repository.
+
+The tests can be run through Visual Studio Code.  Advanced users may also use the `codeql test run` command.
+
+## Contributing to the tests
+
+We are keen to have unit tests for all of our QL code.
+
+Every query in `cpp/ql/src` (outside of `cpp/ql/src/experimental`) should have a test in the corresponding subdirectory of `cpp/ql/test/query-tests`. At a minimum, each query test shall contain one case that should be detected by the query, and one related case that should not.
```

Comment:
Is the requirement that it _should_ (but not necessarily _is_) detected by the query? Or is the requirement that it _should_ (and _is_) detected by the query?

Sorry for being pedantic. This is a great write-up!

Reply:
I personally prefer the former - for example if the query is broken and doesn't detect anything, I'd definitely like to have a test case showing what it should (but doesn't) detect.